project-api-tests/src/test/java/org/ballerina/projectapi/BalToolTest.java

```diff
@@ -0,0 +1,870 @@
+        }
+
+        // check for tool availability in local env
+        balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.REMOVE).balToolsTomlPath);
+        toolOpt = BalToolsManifestBuilder.from(balToolsToml).build().getTool(toolId, specificVersion);
+        if (toolOpt.isPresent()) {
+            Assert.fail("Tool '" + toolIdAndVersion + "' is available in bal-tools.toml");
+        }
+
+        Path toolVersionCachePath = toolEnvironments.get(ToolSubCommand.REMOVE).centralCachePath
+                .resolve(Path.of(orgName, packageName, specificVersion));
+        if (Files.exists(toolVersionCachePath)) {
+            Assert.fail("Tool '" + toolIdAndVersion + "' is available in local cache");
+        }
+    }
+
+    @Test(description = "Remove all tool versions",
+            dependsOnMethods = {"testRemoveASpecificToolVersion", "testRemoveToolVersionWithIncompatibleDistribution"},
+            groups = {"remove"})
+    public void testRemoveAllToolVersions() throws IOException, InterruptedException {
+        // check for tool availability in local env
+        BalToolsToml balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.REMOVE).balToolsTomlPath);
+        Optional<BalToolsManifest.Tool> toolOpt = BalToolsManifestBuilder.from(balToolsToml).build()
+                .getActiveTool(toolId);
+        if (toolOpt.isEmpty()) {
+            Assert.fail("Tool '" + toolId + "' is not available to be removed");
+        }
+
+        Process cmdExec = executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("remove", toolId)),
+                toolEnvironments.get(ToolSubCommand.REMOVE).envVariables);
+        String cmdErrors = getString(cmdExec.getErrorStream());
+        if (!cmdErrors.isEmpty()) {
+            Assert.fail(OUTPUT_CONTAIN_ERRORS + cmdErrors);
+        }
+
+        // Validate the command output
+        String cmdOutput = getString(cmdExec.getInputStream());
+        String expectedOutput = readExpectedCmdOutsAsString("tool-remove-all.txt");
+        if (!cmdOutput.contains(expectedOutput)) {
+            Assert.fail(OUTPUT_NOT_CONTAINS_EXP_MSG + expectedOutput + "\nactual output:" + cmdOutput);
+        }
+
+        // check for tool availability in local env
+        balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.REMOVE).balToolsTomlPath);
+        toolOpt = BalToolsManifestBuilder.from(balToolsToml).build().getActiveTool(toolId);
+        if (toolOpt.isPresent()) {
+            Assert.fail("Tool " + toolId + " is available in bal-tools.toml");
+        }
+
+        Path toolVersionCachePath = toolEnvironments.get(ToolSubCommand.REMOVE).centralCachePath
+                .resolve(Path.of(orgName, packageName));
+        if (Files.exists(toolVersionCachePath)) {
+            Assert.fail("Tool '" + toolId + "' is available in local cache");
```

Comment:
Let's execute the command also and verify it. Applies to other remove tests as well.

Reply:
Fixed with 0b92fd97f16524ed01af439c88ff2b462485cd34